expr/parse.go

```diff
@@ -209,14 +229,14 @@ func Parse(e string, piped bool) (*expr, string, error) {
 	// if we are in a sub-call to Parse (such as the Parse() called in step 2/3/4) we should not try to read upcoming
 	// pipes, but rather have the top-level consume them all.
 
-	for !piped && e != "" {
+	for !pCtx.Piped && e != "" {
 		e = skipWhitespace(e)
 		if e == "" || e[0] != '|' {
 			break
 		}
 		e = e[1:]
 		var nextExp *expr
-		nextExp, e, err = Parse(e, true)
+		nextExp, e, err = Parse(e, ParseContext{Piped: true, IsFullArg: false})
```

Comment:
a comment here with an example about why IsFullArg is false here, would be useful.

Reply:
Added details about how a number can't be an input into a pipe in the comment documenting the  `CanParseAsNumber` in https://github.com/grafana/metrictank/commit/21418b79494179b1834262ad51ef2595787b17e2